modules/stereo/src/stereo_binary_bm.cpp

```diff
@@ -46,693 +46,481 @@
+                    parSumsIntensityImage[1].create(left0.rows, left0.cols,CV_32SC4);
+                    Integral[0].create(left0.rows,left0.cols,CV_32SC4);
+                    Integral[1].create(left0.rows,left0.cols,CV_32SC4);
+                    integral(left, parSumsIntensityImage[0],CV_32S);
+                    integral(right, parSumsIntensityImage[1],CV_32S);
+                    imageMeanKernelSize(parSumsIntensityImage[0], params.kernelSize,Integral[0]);
+                    imageMeanKernelSize(parSumsIntensityImage[1], params.kernelSize, Integral[1]);
+                    modifiedCensusTransform(left,right,params.kernelSize,censusImage[0],censusImage[1],CV_MEAN_VARIATION,0,Integral[0], Integral[1]);
+                }
+                else if(params.kernelType == CV_STAR_KERNEL)
+                {
+                    starCensusTransform(left,right,params.kernelSize,censusImage[0],censusImage[1]);
+                }
+                hammingDistanceBlockMatching(censusImage[0], censusImage[1], hammingDistance);
+                costGathering(hammingDistance, partialSumsLR);
+                blockAgregation(partialSumsLR, params.agregationWindowSize, agregatedHammingLRCost);
+                dispartyMapFormation(agregatedHammingLRCost, disp0, 3);
+                Median1x9Filter(disp0, disp0);
+                Median9x1Filter(disp0,disp0);
+
+                if(params.regionRemoval == CV_SPECKLE_REMOVAL_AVG_ALGORITHM)
+                {
+                    smallRegionRemoval(disp0,params.speckleWindowSize,disp0);
+                }
+                else if(params.regionRemoval == CV_SPECKLE_REMOVAL_ALGORITHM)
+                {
+                    if (params.speckleRange >= 0 && params.speckleWindowSize > 0)
+                        filterSpeckles(disp0, FILTERED, params.speckleWindowSize, params.speckleRange, slidingSumBuf);
+                }
+            }
+            int getAgregationWindowSize() const { return params.agregationWindowSize;}
+            void setAgregationWindowSize(int value = 9) { params.agregationWindowSize = value;}
+
+            int getBinaryKernelType() const { return params.kernelType;}
+            void setBinaryKernelType(int value = CV_MODIFIED_CENSUS_TRANSFORM) { params.kernelType = value; }
+
+            int getSpekleRemovalTechnique() const { return params.regionRemoval;}
+            void setSpekleRemovalTechnique(int factor = CV_SPECKLE_REMOVAL_AVG_ALGORITHM) { params.regionRemoval = factor; }
+
+            bool getUsePrefilter() const { return params.usePrefilter;}
+            void setUsePrefilter(bool value = false) { params.usePrefilter = value;}
+
+            int getScalleFactor() const { return params.scalling;}
+            void setScalleFactor(int factor) {params.scalling = factor; setScallingFactor(factor);}
+
+            int getMinDisparity() const { return params.minDisparity; }
+            void setMinDisparity(int minDisparity) { params.minDisparity = minDisparity; }
```

Comment:
Have you removed all the "contracts" checks (like, `CV_Assert(minDisparity > 0);`?